docs/guides/transactions/running-background-jobs.md

```diff
@@ -118,20 +43,87 @@ namespace StarcounterApplication4
 }
 ```
 
-**Note:** using `thread.Start()` instead of `ThreadPool.QueueUserWorkItem` will lead to the following entries in the Starcounter log and the shutdown will take longer time.
+## Basic Information About Scheduling
+
+Each Starcounter scheduler has a queue of tasks that are supposed to be run on this scheduler. Tasks are picked from the queue and executed. To put a task in a queue, the `Scheduling.RunTask` should be used. When scheduling a task, you can specify the scheduler number, and if the thread should wait for the task to be picked by scheduler and completed. Here is the signature of the `Scheduling.RunTask`:
 
 ```csharp
-Thread foreverThread = new Thread(new ThreadStart(RunForever));
-foreverThread.Start();
+Task RunTask(
+    Action action,
+    Byte schedulerId = StarcounterEnvironment.InvalidSchedulerId)
+```
+
+where:
+
+* `Action action`: procedure to execute on scheduler.
+* `Byte schedulerId = StarcounterEnvironment.InvalidSchedulerId`: optional parameter to select the scheduler, on which the `action` is going to run.
+
+To make the `Action` execute synchronously, use the `Wait` method:
+
+```csharp
+Scheduling.RunTask(() => { }).Wait()
 ```
 
-> 20150928T084607 Warning sc://chrhol-pc/personal Starcounter.Server - User code process takes longer than expected to exit. \(, PID=1271188, Database=default\)
+To determine if current thread is on scheduler call `StarcounterEnvironment.IsOnScheduler()`. To determine the amount of schedulers in your database call `StarcounterEnvironment.SchedulerCount`. To get current scheduler id call `StarcounterEnvironment.CurrentSchedulerId` \(in case if calling thread is not on Starcounter scheduler the value `StarcounterEnvironment.InvalidSchedulerId` is returned\).
+
+## Task scheduling and async/await
+
+Async/await programming in Starcounter is the same as in standard .NET. The only difference is that if there is a Starcounter-related code (SQL, transactions, etc.) that comes after the `await` statement - such code should be wrapped into a delegate given to `Scheduling.RunTask`. Below is an example demonstrating this. It uses delayed `Response`. Note also the use of `Db.TransactAsync` - it's needed so that the Starcounter scheduler does not create additional threads.
```

Comment:
"if there is a Starcounter-related code" -> "if there's Starcounter-related code"